libcxx/include/string

```diff
@@ -2169,6 +2177,7 @@ private:
         __alloc_ = __str.__alloc_;
       } else {
         __annotate_delete();
+        auto __guard = std::__make_scope_guard(__annotate_new_size(*this));
```

Comment:
Was there an exception-safety problem with the code previously? I don't see it.

Actually, with the new code, don't we incorrectly call `annotate_new` if we failed to allocate memory on line 2182?

I think we talked about just this when we spoke last week but somehow I can't remember the details of that discussion.

Reply:
I think so. The string hasn't been modified if we throw in line 2182, so with the old code we've just removed the annotation but haven't added it back anywhere.

Reply:
Ah, yes, that was it.